In [1]:
import pandas as pd
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/googleplaystore.csv"
df = pd.read_csv(url)

Pregunta A: El scraper cometió errores y guardó la misma app varias veces. Escriban una línea de código para identificar si hay nombres duplicados en la columna App (df.duplicated(subset=['App']).sum()). Si los hay, apliquen .drop_duplicates(subset=['App'], keep='first', inplace=True) para quedarse solo con la primera aparición de cada aplicación

In [2]:
res = df.duplicated(subset=['App']).sum()
print("Cantidad de duplicados: ", res)

Cantidad de duplicados:  1181


In [3]:
## Si hay registros duplicados
df.drop_duplicates(subset=['App'], keep='first', inplace=True)

res = df.duplicated(subset=['App']).sum()
print("Cantidad de duplicados: ", res)

Cantidad de duplicados:  0


¿Cuántos registros duplicados purgaron? 

Fueron en total 1181 registros que fueron eliminados

Pregunta B: La columna Installs tiene texto como "1,000,000+". Un algoritmo no puede procesar eso.

a. Escriban una instrucción encadenando el método .str.replace() para eliminar el símbolo + y otro para eliminar la coma ,. (Ojo: utilicen regex=False o escapen el símbolo más \+).

In [4]:
df['Installs'] = df['Installs'].str.replace('+',"",regex=False)
df['Installs'] = df['Installs'].str.replace(',',"",regex=False)

df['Installs'] = pd.to_numeric(df['Installs'],errors = 'coerce') ##There some texts like Free that generate erros, so i passed as NaN cause theres no any value

print(df['Installs'])

0           10000.0
1          500000.0
2         5000000.0
3        50000000.0
4          100000.0
            ...    
10836        5000.0
10837         100.0
10838        1000.0
10839        1000.0
10840    10000000.0
Name: Installs, Length: 9660, dtype: float64


b. Al final de la cadena, utilicen .astype(int) (o float si tienen NaNs) para convertir la columna.

In [5]:
df['Installs'] = df['Installs'].astype(float)
print(df['Installs'].dtype)

float64


Comprueben el éxito ejecutando df['Installs'].mean(). 
¿Cuál es el promedio real de descargas?

El promedio de las descargas son de 7,777,506.73 

In [6]:
average = df['Installs'].mean()
print("Downloads Average: ", average)

Downloads Average:  7777506.732270421


Pregunta C: La columna Price tiene valores como "$4.99" y "0". Apliquen la misma lógica arquitectónica que en la Pregunta B: eliminen el símbolo de dólar $ y conviertan la columna usando .astype(float).

In [7]:
##Same process - delete $ sign and convert to int or float
df['Price'] = df['Price'].str.replace('$','',regex=False)
df['Price'] = df['Price'].str.replace('.','',regex=False)

df['Price'] = pd.to_numeric(df['Price'], errors='coerce')

df['Price'] = df['Price'].astype(float)

print(df['Price'].dtype)

float64


Pregunta D: Ahora que el precio es numérico, ejecuten df['Price'].max(). Notarán que la app más cara cuesta alrededor de 400 dólares.

In [8]:
print("Price of the max app : ", df['Price'].max())

Price of the max app :  40000.0


a. Usen un filtro booleano (df[df['Price'] > 200]) para descubrir cuáles son estas apps. Verán que son aplicaciones basura o de broma.

In [9]:
print("Apps : ", df[df['Price'] > 20000])

Apps :                                      App   Category  Rating Reviews  Size  \
4197             most expensive app (H)     FAMILY     4.3       6  1.5M   
4362                         💎 I'm rich  LIFESTYLE     3.8     718   26M   
4367           I'm Rich - Trump Edition  LIFESTYLE     3.6     275  7.3M   
5351                          I am rich  LIFESTYLE     3.8    3547  1.8M   
5354                     I am Rich Plus     FAMILY     4.0     856  8.7M   
5355                      I am rich VIP  LIFESTYLE     3.8     411  2.6M   
5356                  I Am Rich Premium    FINANCE     4.1    1867  4.7M   
5357                I am extremely Rich  LIFESTYLE     2.9      41  2.9M   
5358                         I am Rich!    FINANCE     3.8      93   22M   
5359                 I am rich(premium)    FINANCE     3.5     472  965k   
5362                      I Am Rich Pro     FAMILY     4.4     201  2.7M   
5364     I am rich (Most expensive app)    FINANCE     4.1     129  2.7M   
5366

b. Si metemos estas apps al modelo, la red neuronal creerá que vender a 400 dólares es un modelo de negocio viable. Sobrescriban el DataFrame conservando únicamente las apps que cuesten menos de 50 dólares (df = df[df['Price'] < 50]).

In [10]:
df = df[df['Price'] < 5000]
print(df.size)


125268


c.Guarden su dataset final usando df.to_csv('playstore_limpio.csv', index=False).

In [11]:
df.to_csv('playstore_limpio.csv', index= False)